src/services/anchor/index.ts

```diff
@@ -194,10 +218,23 @@ export async function sep24First(sessionParams: IAnchorSessionParams): Promise<I
   const { sep24Url } = tomlValues;
 
   // at this stage, assetCode should be defined, if the config is consistent.
-  const sep24Params = new URLSearchParams({
-    asset_code: sessionParams.tokenConfig.stellarAsset.code.string,
-    amount: sessionParams.offrampAmount,
-  });
+  let sep24Params;
+
+  if (sessionParams.tokenConfig.stellarAsset.code.string === 'ARS\0') {
+    if (!masterPublic) {
+      throw new Error('Master must be defined at this point.');
+    }
+    sep24Params = new URLSearchParams({
+      asset_code: 'ARS',
+      amount: sessionParams.offrampAmount,
+      account: masterPublic, // Since we signed with the master from the service, we need to specify the corresponding public here
```

Comment:
Shouldn't this be the ephemeral account instead? The SEP-24 standard defines `account` as:

> The Stellar or muxed account the client will use as the source of the withdrawal payment to the anchor. Defaults to the account authenticated via SEP-10 if not specified.

Reply:
Yes, you are right. But it doesn't work with it. It only works with the one that did the `sep-10`. I missed that part of the docs and found this by trial an error.